Schedule.API/Services/Procedures/RecommendationService.cs

```diff
@@ -253,5 +254,94 @@ private IEnumerable<EquipmentRelocationDto> FindCommonEquipmentRelocationAppoint
+            foreach (Shift shift in shiftsForSourceRoom)
+            {
+                List<Shift> shiftList = new List<Shift>();
+                shiftList.Add(shift);
+                IEnumerable<RecommendationDto> recommendationsSourceRoom = FindRecommendationsInShifts(shiftList, 16);
+                if (recommendationsSourceRoom.Count() == 16)
+                {
+                    availableShiftsSource.Add(shift);
+                }
+            }
+            
+            foreach (Shift shift in availableShiftsSource)
+            {
+                RoomRenovationDto roomRenovationDto = new RoomRenovationDto()
+                {
+                    FirstRoomId = dto.FirstRoomId,
+                    SecondRoomId = dto.SecondRoomId,
+                    TimeInterval = shift.TimeInterval
+                };
+                appointments.Add(roomRenovationDto);
+            }
+            return appointments;
+        }
+
+        private IEnumerable<RoomRenovationDto> ComplexRenovationSearch(SchedulingDto dto)
+        {
+            List<RoomRenovationDto> appointments = new List<RoomRenovationDto>();
+            List<Shift> availableShiftsSource = new List<Shift>();
+            List<Shift> availableShiftsDestination = new List<Shift>();
+            IEnumerable<Shift> shiftsForDestinationRoom = _shiftWrapper.Repository.GetShiftsByRoomID(dto.SecondRoomId);
+            IEnumerable<Shift> shiftsForSourceRoom = _shiftWrapper.Repository.GetShiftsByRoomID(dto.FirstRoomId);
+            foreach (Shift shift in shiftsForSourceRoom)
+            {
+                List<Shift> shiftList = new List<Shift>();
+                shiftList.Add(shift);
+                IEnumerable<RecommendationDto> recommendationsSourceRoom = FindRecommendationsInShifts(shiftList, 16);
+                if (recommendationsSourceRoom.Count() == 16)
+                {
+                    availableShiftsSource.Add(shift);
+                }
+            }
+            foreach (Shift shift in shiftsForDestinationRoom)
+            {
+                List<Shift> shiftList = new List<Shift>();
+                shiftList.Add(shift);
+                IEnumerable<RecommendationDto> recommendationsDestinationRoom = FindRecommendationsInShifts(shiftList, 16);
+                if (recommendationsDestinationRoom.Count() == 16)
+                {
+                    availableShiftsDestination.Add(shift);
+                }
+            }
+
+            List<TimeInterval> availableTimeIntervals = new List<TimeInterval>();
+            foreach (Shift shiftSource in availableShiftsSource)
+            {
+                foreach (Shift shiftDest in availableShiftsDestination)
+                {
+                    if (shiftDest.TimeInterval.Start == shiftSource.TimeInterval.Start && shiftDest.TimeInterval.End == shiftSource.TimeInterval.End)
+                    {
+                        availableTimeIntervals.Add(shiftDest.TimeInterval);
+                    }
```

Comment:
why do you look at strict boundaries

Reply:
I had to check if dates are same. It is fixed now.